core/ir/isa_regdeps/disassemble.c

```diff
@@ -0,0 +1,77 @@
+/* **********************************************************
+ * Copyright (c) 2024 Google, Inc.  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of VMware, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#include "../globals.h"
+#include "disassemble.h"
+
+/* DR_ISA_REGDEPS instruction encodings can be at most 16 bytes and are 4 byte aligned,
+ * hence we can have at most 4 lines of 4 bytes each.
```

Comment:
Why are we limited to 4 bytes per line?  It doesn't seem great to have a 16-byte encoding with the first line with 4 encoding bytes and the actual disasm and then 3 following lines each with just 4 bytes of more encoding.  Seems better to either put the whole 16 bytes one one line or if that is too much empty space and 90% of the instructions are at <=8 bytes then put 8 bytes on each of 2 lines?

Reply:
I thought 4 bytes per line was more similar, in terms of layout, to our description and examples of the encoding in
https://github.com/DynamoRIO/dynamorio/blob/a3411cb0eff4c6b6c1bff2dee37906baf8e7b457/core/ir/isa_regdeps/encoding_common.h#L84
since everything is in 4 byte chunks.

But we are not limited by it, I'm fine with having 8 bytes per line (maybe divided into two 4 byte words?) so we can stay within 2 lines instead of 4.

Reply:
Done. Went with 2 lines max, 8 bytes each, split in two 4 byte words (if the second/fourth word is present).

Reply:
SGTM